### ISSUE TITLE
undefined method `block_is_haml?' for

### ISSUE THREAD

**Issue by gabrieletassoni** [excid3/simple_calendar#118]
Hello,
I'm getting the error in the subject whenever I use, in the view, this code:

```
<%= month_calendar events: Event.all do |date, events| %>
  <%= date %>

  <% events.each do |event| %>
    <div style="background-color: <%=event.subscribers.pluck(:user_id).include?(current_user.id) ? "red" : "lightgray"%>">
      <%= link_to(event.name, event_path(event.id)) %>
    </div>
  <% end %>
<% end %>
```

In particular the error is (using rails 4.2):

```
Showing /Users/gabrieletassoni/.rvm/gems/ruby-2.2.2/gems/simple_calendar-2.0.3/app/views/simple_calendar/_month_calendar.html.erb where line #20 raised:

undefined method `block_is_haml?' for #<#<Class:0x007ffbe2a1b7e8>:0x007ffbe36b7420>
```

And the Event.all results in (using in .inspect to show it in the log):

```
#<ActiveRecord::Relation [#<Event id: 3, logo_file_name: "NYP_Spellman-Classroom.jpg", logo_content_type: "image/jpeg", logo_file_size: 277771, logo_updated_at: "2016-01-29 11:44:56", title: "Meeting tecnico", description: "<p><strong>Siamo ad invitarvi presso i nostri Loca...", recurrence: nil, user_id: 1, created_at: "2016-01-29 11:44:56", updated_at: "2016-01-29 11:44:56", published: true, start_time: "2016-02-03 12:43:00", end_time: "2016-02-03 13:43:00", color: "ad8aad">, #<Event id: 4, logo_file_name: nil, logo_content_type: nil, logo_file_size: nil, logo_updated_at: nil, title: "Meeting commerciale", description: "<p>Presentazione a tutti quellic he vogliono partc...", recurrence: nil, user_id: 1, created_at: "2016-01-29 11:46:48", updated_at: "2016-01-29 11:46:48", published: true, start_time: "2016-02-06 14:46:00", end_time: "2016-02-07 13:46:00", color: "42822c">]>
```

What can I do to resolve/better investigate this?

Thank you very much,
Gabriele

**Comment by gabrieletassoni** [comment]
Investigating, it throws the same error is thrown without "events: " also, thus I think there's something I'm misinterpreting rather than a possibile bug, I work on it and report back if I can understand where I'm failing at, at least for future reference.

**Comment by excid3** [comment]
Are you using Haml anywhere in your app? One of the things that we ran into previously was that Haml causes odd rendering if you mix it with Erb and added the check to see if the block is haml. From my testing so far, it's been working, but could certainly need some more tweaking.

Your example there looks like it's all in Erb which makes me curious as to what might have triggered that Haml stuff.

**Comment by gabrieletassoni** [comment]
Hi,
well, not directly, maybe devise or rails_admin uses it… I may check.

**Comment by mikeholford** [comment]
I had the same but I removed my Haml gem as I was luckily not using it yet

**Comment by excid3** [comment]
Interesting. The code for this that's getting tripped up is the following: 

```ruby
              <% if defined?(Haml) && block_is_haml?(block) %>
```

Which only runs if Haml is already setup. I wonder if this method no longer exists in a newer version of Haml? (I'm not super familiar with it).

**Comment by excid3** [comment]
Just figured out what was wrong and fixed this! The helper that contains the `block_is_haml?` method doesn't get included unless you're rendering a haml template now. Pretty sure this used to be different, but at least that's the case now in the latest version. This should be all fixed in 2.0.6 if you update to that.

**Comment by gabrieletassoni** [comment]
Many thanks. ^_^
